scripts/createYoeoLabelsTORSO.py

```diff
@@ -127,134 +121,129 @@ def range_limited_float_type_0_to_1(arg):
-        # Generate segmentations in correct format
-        seg_path = os.path.join(os.path.dirname(yamlfile), "segmentations", os.path.splitext(img_name)[0] + ".png")
-        seg_in = cv2.imread(seg_path)
-        if seg_in is not None:
-            mask = np.zeros(seg_in.shape[:2], dtype=np.uint8)
-            mask += ((seg_in == (127, 127, 127)).all(axis=2)).astype(np.uint8)  # Lines
-            mask += (((seg_in == (254, 254, 254)).all(axis=2)).astype(np.uint8) * 2)  # Field
-            seg_out = np.zeros(seg_in.shape, dtype=np.uint8)
-            seg_out[..., 0] = mask
-            seg_out[..., 1] = mask
-            seg_out[..., 2] = mask
-            cv2.imwrite(os.path.join(masks_dir, os.path.splitext(img_name)[0] + ".png"), seg_out)
-        else:
-            print(f"No segmentation found: '{seg_path}'")
-            continue
-
-        images.append(os.path.join(d, "images", img_name))
-        name = os.path.splitext(img_name)[0]  # Remove file extension
-        imgwidth = frame['width']
-        imgheight = frame['height']
-        annotations = []
-
-        for annotation in frame['annotations']:
-            # Ignore if blurred or conceiled and should be ignored
-            if not ((args.ignore_blurred and annotation['blurred']) or
-                (args.ignore_conceiled and annotation['conceiled'])):
-
-                if annotation['type'] in CLASSES['bb_classes']:  # Handle bounding boxes
-                    if annotation['in_image']:
-                        min_x = min(map(lambda x: x[0], annotation['vector']))
-                        max_x = max(map(lambda x: x[0], annotation['vector']))
-                        min_y = min(map(lambda x: x[1], annotation['vector']))
-                        max_y = max(map(lambda x: x[1], annotation['vector']))
-
-                        annowidth = max_x - min_x
-                        annoheight = max_y - min_y
-                        relannowidth = annowidth / imgwidth
-                        relannoheight = annoheight / imgheight
-
-                        center_x = min_x + (annowidth / 2)
-                        center_y = min_y + (annoheight / 2)
-                        relcenter_x = center_x / imgwidth
-                        relcenter_y = center_y / imgheight
-
-                        classID = CLASSES['bb_classes'].index(annotation['type'])  # Derive classID from index in predefined classes
-                        annotations.append(f"{classID} {relcenter_x} {relcenter_y} {relannowidth} {relannoheight}")  # Append to store it later
-                    else:  # Annotation is not in image
-                        continue
-                elif annotation['type'] in CLASSES['segmentation_classes']:  # Handle segmentations
+# Defaults
+create_symlinks = False
+dataset_dir = args.dataset_dir
+destination_dir = args.dataset_dir
+image_names = []  # Collect image paths for train/test split
+
+# Overwrite defaults, if destination path is given
+if args.destination_dir:
```

Comment:
What is the intended use case of a different destination without the symlinks? In the non symlink case no data is transfered. I would normally assume that the data is copied if no symlink is used. One could also use the symlink implicitly if a different destination is set

Reply:
You could still choose a custom `destination-dir`, and read the images from the original `dataset-dir` place without symlink. Although, I broke this by setting `destination-dir` to the images path...
(Initially, I implemented `destination-dir` and symlinks as a combined function, then I separated it, then I broke it.)

Do we want to support this use-case?

Reply:
Also, the file tree would be kind of weird in this case.
Something like:

`
torso/
   train/
      images/
      segmentations/
      annotations.yaml

destination_dir/
    train/
        labels/
        yoeo_labels/
`

I would therefore always use the symlink when the destination dir differs.